File: Course_2-Data_Structures/Week-4/Excercise_Challenges/1_phone_book/phone_book.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

int main(void)
{
    int n;
    cin >> n;

    map<int, string> m;
    string a, c;
    int b;

    for(int i = 0; i < n; ++i){
        cin >> a;

        if(a == "add"){
            cin >> b >> c;
            m[b] = c;
        }
        else if(a == "del"){
            cin >> b;

            if(m.count(b)){
                auto it = m.find(b);
                m.erase(it);
            }                
        }
        else if(a == "find"){
            cin >> b;
            if(m.count(b))
                cout << m[b] << endl;
            else
                cout << "not found" << endl;
        }
    }

    return 0;
}

File: Course_2-Data_Structures/Week-4/Excercise_Challenges/1_phone_book/phone_book.py
# python3

class PhoneBook:
    def __init__(self):
        self.n = 0
        self.phone_list = {}

    def read_queries(self):
        self.n = int(input())

        for _ in range(self.n):
            data = input().split()

            if data[0] == 'add':
                self.phone_list[data[1]] = data[2]
            elif data[0] == 'find':
                if data[1] in self.phone_list:
                    print(self.phone_list[data[1]])
                else:
                    print('not found')
            elif data[0] == 'del':
                if data[1] in self.phone_list:
                    self.phone_list.pop(data[1])
        
    def solve(self):
        self.read_queries()
    
if __name__ == '__main__':
    obj = PhoneBook()
    obj.solve()

